pact/coin-contract/coin-faucet.pact

```diff
@@ -0,0 +1,94 @@
+(module coin-faucet FAUCET-GOVERNANCE
+
+  "'coin-faucet' represents Kadena's Coin Faucet Contract."
+
+  (defcap FAUCET-GOVERNANCE () true)
+
+  (use coin)
+
+  ; --------------------------------------------------------------------------
+  ; Schemas and Tables
+  ; --------------------------------------------------------------------------
+
+  (defschema history
+    @doc "Table to record of the behavior of addresses. Last transaction time,   \
+    \ total coins earned, and total coins returned are recorded per transaction. "
+    last-tx-time:time
+    total-coins-earned:decimal
+    total-coins-returned:decimal
+    )
+
+  (deftable history-table: {history})
+
+  ; --------------------------------------------------------------------------
+  ; Capabilities
+  ; --------------------------------------------------------------------------
+
+  (defcap HISTORY ()
+    "Enforces faucet-guard when looking up history"
+    (enforce-guard (faucet-guard)))
+
+  ; --------------------------------------------------------------------------
+  ; Constants
+  ; --------------------------------------------------------------------------
+
+  (defconst FAUCET_ACCOUNT 'faucet-account)
+  (defconst COIN_PER_REQUEST 1.0)
+  (defconst WAIT_TIME_PER_REQUEST 3600.0)
+  (defconst EPOCH  (time "1970-01-01T00:00:00Z"))
+
+  ; --------------------------------------------------------------------------
+  ; Coin Faucet Contract
+  ; --------------------------------------------------------------------------
+
+  (defun faucet-guard:guard () (create-module-guard 'faucet-admin))
+
+  (defun request-coin:string (address address-guard)
```

Comment:
Can we extend this to allow people to request more than one coin?  I know doesn't seem like a very significant difference, but psychologically it feels better to have a few more coins.  It also scales better with evolving transaction fees.  A limit of 10 - 20 coins seems like a reasonable start.

Reply:
Changed request-coin to allow up to 20 maximum coins per request.